tests/src/rules/no-duplicates.js

```diff
@@ -129,6 +129,20 @@ ruleTester.run('no-duplicates', rule, {
       errors: ['\'./foo\' imported multiple times.', '\'./foo\' imported multiple times.'],
     }),
 
+    // #2347: duplicate identifiers should be removed
+    test({
+      code: "import {a} from './foo'; import { a } from './foo'",
+      output: "import {a} from './foo'; ",
+      errors: ['\'./foo\' imported multiple times.', '\'./foo\' imported multiple times.'],
+    }),
+
+    // #2347: duplicate identifiers should be removed
+    test({
+      code: "import {a,b} from './foo'; import { b, c } from './foo'; import {b,c,d} from './foo'",
+      output: "import {a,b, c,d} from './foo';  ",
+      errors: ['\'./foo\' imported multiple times.', '\'./foo\' imported multiple times.', '\'./foo\' imported multiple times.'],
+    }),
+
```

Comment:
I did build the plugin with my changes incorporated and tested it out manually to verify it works.

These tests should pass but they are failing due to parsing errors:

```
  2236 passing (7s)
  2 failing

  1) no-duplicates invalid import {a} from './foo'; import { a } from './foo':

      AssertionError [ERR_ASSERTION] [ERR_ASSERTION]: A fatal parsing error occurred: Parsing error: Identifier 'a' has already been declared
      + expected - actual

      -false
      +true
      
      at runRuleForItem (node_modules/eslint/lib/rule-tester/rule-tester.js:693:13)
      at testInvalidTemplate (node_modules/eslint/lib/rule-tester/rule-tester.js:800:28)
      at Context.<anonymous> (node_modules/eslint/lib/rule-tester/rule-tester.js:1039:29)
      at processImmediate (node:internal/timers:466:21)

  2) no-duplicates invalid import {a,b} from './foo'; import { b, c } from './foo'; import {b,c,d} from './foo':

      AssertionError [ERR_ASSERTION] [ERR_ASSERTION]: A fatal parsing error occurred: Parsing error: Identifier 'b' has already been declared
      + expected - actual

      -false
      +true
      
      at runRuleForItem (node_modules/eslint/lib/rule-tester/rule-tester.js:693:13)
      at testInvalidTemplate (node_modules/eslint/lib/rule-tester/rule-tester.js:800:28)
      at Context.<anonymous> (node_modules/eslint/lib/rule-tester/rule-tester.js:1039:29)
      at processImmediate (node:internal/timers:466:21)
```

I'm not sure how to proceed here?

Reply:
Hmm, that's a really good question. If duplicate identifiers aren't parseable at all, then i'm not sure how the plugin would be capable of linting them.

You could play with different parsers with the `parser: ` option, and see if maybe one (or all) of the babel or TS parsers can handle it?

Reply:
That worked, thanks!